RFC0032.md

```diff
@@ -0,0 +1,112 @@
+
+* Sending notifications via email (or other channels) to a designated user. Application users may wish to be notified of completion of their application, receive periodic progress reports, or be notified of a problem that merits attention. PMIx itself includes support for some of the more popular channels - requests for unsupported channels are referred to the host RM for processing, with an error returned if the requested channel is not available in the host environment
+
+* Outputting tagged log messages to stdout or stderr of the application, or a connected tool. Where supported, an alternative output stream (possibly directed to a dedicated log file) may be specified. Messages may be tagged (via the PMIX\_LOG\_TAG\_OUTPUT} attribute) as flowing via the PMIx\_Log API to differentiate them from the application's normal output. In addition, messages may be timestamped (PMIX\_LOG\_TIMESTAMP\_OUTPUT) or output in XML format (PMIX\_LOG\_XML\_OUTPUT)
+
+* Storing status updates in the job record using the PMIX\_LOG\_JOB\_RECORD attribute. Resource managers nearly always maintain a record of the jobs they schedule and execute. This includes information on the time spent waiting for allocation, priority of the request, identity of the requestor, name/path of the executable and/or job script, etc. Historically, users have had to record status information on their application (e.g., computational progress) in files which are subsequently stored in persistent storage. PMIx\_Log offers the option (where supported) of injecting such status reports directly into the job record, thus providing a single, time sequential record of the job's execution.
+
+  Note that system libraries can also use this feature to record job-affecting events (e.g., network failures) that might have impacted the application during its execution, perhaps linking them to more detailed information stored in a RAS database.
+
+* Storing state information in a global key-value datastore. The prior use-cases all involve one-way output of data - i.e., the caller can log the data, but cannot later retrieve it. However, there are times when an application, tool, or SMS element may wish to store information in a global key-value datastore that can be searched by external agents, or be retrieved by the caller itself at some later time. For example, an SMS element may wish to store state information in a persistent place for retrieval upon recovery from a failure. Use of the PMIx\_Publish API might, at first glance, appear to meet this need. However, PMIx_Publish is focused on inter/intra-application data exchange - it therefore does not guarantee persistence across (for example) an RM failure.
+
```

Comment:
Still not clear to me how I do this?   how do I make a log that specifically goes to the datastore?  If I do this:


+PMIX_INFO_LOAD(&data[0], PMIX_LOG_STDOUT, "my message string", PMIX_STRING);
+PMIX_INFO_LOAD(&directive[0], PMIX_LOG_GLOBAL_DATASTORE, 1, PMIX_BOOL);  
(totally guessing on the types here because they need to be documented)

Is that going to write "my message string" to stdout or is it going to create a PMIx info in the global data store with PMIX_LOG_STDOUT set to "my message string"?

Reply:
I'll try to better clarify the difference between the data and directives fields.